akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1764,11 +1764,14 @@ private[stream] object Collect {
       def onPush(): Unit = {
         if (buffer.isFull)
           onPushWhenBufferFull()
-        else {
+        else if (buffer.isEmpty && !isTimerActive(timerName)) {
+          // ONLY schedule the timer if the buffer is empty, as an initial condition for the timing mechanism.
+          // Otherwise, scheduling the timer will starve subsequent `onPull` callbacks,
+          // even though elements are already overdue. The starvation exacerbates when the delay duration increases.
+          grabAndPull()
+          scheduleOnce(timerName, d)
+        } else {
           grabAndPull()
```

Comment:
Thank you for a really thorough investigation of the issue. We really appreciate that.

I see a potential problem in this `else` branch where we do not schedule a timer. If the `pullCondition` is false (buffer is full, for example) the `grabAndPull` will not pull, and then this stage might get stuck - it will not receive `onPull` and neither `onTimer`.

For an alternative fix, have you considered changing the `onPush` where the duration for the `scheduleOnce` is `nextElementWaitTime().millis` if the buffer is not empty, and `d` otherwise? This would directly address the current issue, which is scheduling the timer for a full duration, even if there are elements already in the buffer.

Reply:
Thanks for the review. I'll look into that and get back to you.

Reply:
Hi @2m, thanks again for the review comments. Sorry that I didn't get to address it in the past a few weeks (a bit hectic.)

If my understanding of the code is correct, in the case that the buffer is not empty, and that there's no active timer, the `onPull` function should schedule a timer based on `nextElementWaitTime()` already:
https://github.com/akka/akka/blob/c4bf933b3c163061e195451fec722334a592d0cf/akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala#L1790-L1797

Alternatively, if `onPull` is called when a timer is active (so that it does nothing), then when that timer is fired, it will set up another timer if necessary:
https://github.com/akka/akka/blob/c4bf933b3c163061e195451fec722334a592d0cf/akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala#L1818-L1822

Are there any other edge cases where the stage is pulled, but somehow no timer is set, causing the element to be stuck in the buffer? 

With that said, I did update `onPush` according to your suggestions so that it should cover all the corner cases.

Reply:
Thanks for getting back!

After updating `onPush`, I do not see any problems with the current state of this PR.